pccommon/pccommon/redis.py

```diff
@@ -74,6 +79,25 @@ async def connect_to_redis(app: FastAPI) -> None:
     app.state.redis_back_pressure_script_hash = back_pressure_script_hash
 
 
+async def reregister_scripts(request: Request) -> None:
+    """Attempt to reregister lua scripts if raised as missing."""
+
+    # Avoid multiple threads trying to reregister scripts at once
+    with redis_script_lock:
+        state: State = request.app.state
+        r: Redis = state.redis
+
+        scripts = [
+            (state.redis_rate_limit_script_hash, rate_limit_lua_script),
+            (state.redis_back_pressure_script_hash, back_pressure_lua_script),
+        ]
+
+        exists = await r.script_exists(*[s[0] for s in scripts])
+        for script_exists, idx in enumerate(exists):
+            if not script_exists:
+                await r.script_load(scripts[idx][1])
```

Comment:
does `r.script_load` always return the same hash for the script, so the script hash will be the same? Or does e.g. `state.redis_rate_limit_script_hash` need to be reset to the return value here?

Reply:
It does, the hash is a `SHA1` digest of the script content. The test verifies this by registering, flushing, and then invoking the script with the original hash.